src/InstrumentationEngine/InstructionGraph.cpp

```diff
@@ -811,8 +785,10 @@ HRESULT MicrosoftInstrumentationEngine::CInstructionGraph::InsertBefore(_In_ IIn
     IfNullRetPointer(pInstructionOrig);
     IfNullRetPointer(pInstructionNew);
 
-    CInstruction* pInstrOrig = (CInstruction*)pInstructionOrig;
-    CInstruction* pInstrNew = (CInstruction*)pInstructionNew;
+    CComPtr<CInstruction> pInstrOrig;
+    IfFailRet(pInstructionOrig->QueryInterface(&pInstrOrig));
```

Comment:
>IfFailRet(pInstructionOrig->QueryInterface(&pInstrOrig)); [](start = 4, length = 57)

These QI's will always fail because QI is not implemented for CInstruction. It currently can't be because of diamond inheritance. See #311

Reply:
I updated ```CInstruction::QueryInterface```.